JniBridge.bee.cs

```diff
@@ -0,0 +1,379 @@
+    {
+        var apiGeneratorDir = new NPath("apigenerator");
+        return SetupJarForDirectory(jdk, apiGeneratorDir);
+    }
+
+    static NPath SetupJniBridgeJar(Jdk jdk)
+    {
+        var jnibridgeDir = new NPath("jnibridge");
+        return SetupJarForDirectory(jdk, jnibridgeDir);
+    }
+
+    static NPath SetupJarForDirectory(Jdk jdk, NPath jarDir)
+    {
+        var jarname = jarDir.FileName;
+        var apiGeneratorFiles = jarDir.Files("*.java", true);
+        var classFileDir = new NPath($"artifacts/{jarname}");
+        var apiGeneratorJar = new NPath($"build/{jarname}.jar");
+        var classFiles = jdk.SetupCompilation(classFileDir, apiGeneratorFiles, new NPath[] { jarDir }, new NPath[0]);
+        jdk.SetupJar(new NPath[] { classFileDir }, classFiles, apiGeneratorJar);
+        Backend.Current.AddAliasDependency(jarname, apiGeneratorJar);
+        return apiGeneratorJar;
+    }
+
+    struct SourceGenerationParams
+    {
+        public string platformName;
+        public NPath[] inputJars;
+        public string[] classes;
+    }
+
+    static SourceGenerationParams GetAndroidSourceGenerationParams(NPath sdk)
+    {
+        var androidJar = sdk.Combine("platforms", kAndroidApi, "android.jar");
+        var googlePlayServicesJar = sdk.Combine("extras", "google", "google_play_services_froyo", "libproject", "google-play-services_lib", "libs", "google-play-services.jar");
+
+        return new SourceGenerationParams()
+        {
+            platformName = "android",
+            inputJars = new[] {androidJar, googlePlayServicesJar},
+            classes = kAndroidApiClasses,
+        };
+    }
+
+    static SourceGenerationParams GetMacOSSourceGenerationParams(Jdk jdk)
+    {
+        var rtJar = jdk.JavaHome.Combine("jre", "lib", "rt.jar");
+
+        return new SourceGenerationParams()
+        {
+            platformName = "osx",
+            inputJars = new [] {rtJar},
+            classes = kMacOSClasses,
+        };
+    }
+    static NPath SetupSourceGeneration(Jdk jdk, NPath apiGenerator, NPath jnibridgeJar, SourceGenerationParams genParams)
+    {
+        var destDir = new NPath("artifacts").Combine("generated", genParams.platformName);
+        var actionName = "generate:jnibridge:" + genParams.platformName;
+        var generatedFiles = new NPath[] { destDir.Combine("API.h") };
+
+        var templateFiles = new NPath("templates").Files(new [] { "cpp", "h" });
+        var inputs = new List<NPath>();
+        inputs.AddRange(genParams.inputJars);
+        inputs.Add(jnibridgeJar);
+        inputs.Add(apiGenerator);
+        inputs.AddRange(templateFiles);
+
+        string inputJars = string.Empty;
+        if (genParams.inputJars.Length > 0)
+        {
+            var builder = new StringBuilder();
+            builder.Append('"').Append(genParams.inputJars[0]);
+            for (int i = 1; i < genParams.inputJars.Length; ++i)
+                builder.Append(';').Append(genParams.inputJars[i]);
```

Comment:
; is an OSX specific java separator. It would probably be better to do something similar to what we do in bee:
public static string PathSeparator => HostPlatform.IsWindows ? ";" : ":";
static string CombinePathsWithSeparator(IEnumerable<NPath> paths) => paths.Select(c => c.ToString()).SeparateWith(PathSeparator);

Reply:
This still doesn't build on Windows, there are more OSX specific things. Do we really need JNIBridge to be buildable on Windows?

Reply:
I think - ultimately yes, I mean for Unity's Android player we have the ability to develop on any host OS - Windows, Mac, Linux. JNIBridge shouldn't be any different. Without such ability, it becomes very inconvenient, if you're usual work environment is non Mac.

In any case, no need to add Windows support in this PR, making it compile with Bee is already huge step forward. Windows part can be done later